locator/topology.hh

```diff
@@ -12,37 +12,128 @@
 
 #include <unordered_set>
 #include <unordered_map>
+#include <iostream>
 
 #include <seastar/core/future.hh>
 #include <seastar/core/sstring.hh>
 #include <seastar/util/bool_class.hh>
 
 #include "locator/types.hh"
 #include "inet_address_vectors.hh"
-#include "utils/fb_utilities.hh"
 
 using namespace seastar;
 
 namespace locator {
 
+class topology;
+
+class node : public enable_lw_shared_from_this<node> {
+public:
+    using local = bool_class<struct local_tag>;
+
+private:
+    host_id _host_id;
+    inet_address _endpoint;
+    endpoint_dc_rack _dc_rack;
+
+    // Private state of this instance.
+    // May change across topology generations
+    local _is_local;
```

Comment:
I don't understand what "local" means here. local to the datacenter?

Reply:
I guess it makes sense in view of the term "local quorum". So if that's the answer, the name is okay.

Reply:
"local" here means the local node (previously referred to as fb_utilities::get_broadcast_address), not any node in the same DC as this node, more in the sense of "localhost".

Reply:
renamed to "this_node" in v8